### DISCUSSION TITLE
Writing Branch of LorentzVector objects to a tree

### DISCUSSION THREAD

**Discussion by jkwinter** [scikit-hep/uproot5#567] [General]
Hi, I am trying to convert data from an .lhe format into a ROOT file and I would like to store a numpy array of the four-momentum of each particle as a branch in a tree, for which I am using the scikit-hep LorentzVector class. What type should I specify for the branch when I do mktree(treename,{'branch1' : <type>} title = "")?

**Comment by jpivarski** [comment]
Uproot doesn't (currently) write the TLorentzVector or `ROOT::Math::LorentzVector<ROOT::Math::PxPyPzM4D<double>>` types, so you'll have to write the four components as separate TBranches:

```python
writable_directory.mktree("treename", {"px": np.float64, "py": np.float64, "pz": np.float64, "E": np.float64})
writable_directory["treename"].extend({
    "px": [p4.x for p4 in many_lorentz_vectors],
    "py": [p4.y for p4 in many_lorentz_vectors],
    "pz": [p4.z for p4 in many_lorentz_vectors],
    "E": [p4.t for p4 in many_lorentz_vectors],
})
```

**Reply by jkwinter** [reply]
I see, thank you. Are there plans to allow for these types?

**Reply by jpivarski** [reply]
It would be a reasonable thing to do. If you make it a feature request, it won't be forgotten, though I don't think it will be implemented soon (this month).

→ Unless anybody out there wants to try it! I'll help you through the process, if you do! ←
